src/nvim/autocmd.c

```diff
@@ -1493,6 +1507,77 @@ void aucmd_restbuf(aco_save_T *aco)
   }
 }
 
+/// Schedules an autocommand event, to be executed at the next event-loop tick.
+///
+/// @param event event that occurred
+/// @param fname filename, NULL or empty means use actual file name
+/// @param fname_io filename to use for <afile> on cmdline,
+///                 NULL means use `fname`.
+/// @param group autocmd group ID or AUGROUP_ALL
+/// @param buf Buffer for <abuf>
+/// @param eap Ex command arguments
+void aucmd_defer(event_T event, char *fname, char *fname_io, int group, buf_T *buf, exarg_T *eap,
+                 Object *data)
+{
+  DeferredEvent *evdata = xmalloc(sizeof(DeferredEvent));
+  evdata->event = event;
+  evdata->fname = fname;
+  evdata->fname_io = fname_io;
+  evdata->group = group;
+  evdata->buf = buf->handle;
+  evdata->eap = eap;
+  if (data) {
+    evdata->data = xmalloc(sizeof(Object));
+    *evdata->data = copy_object(*data, NULL);
```

Comment:
Do we need to free `data` here ? oh, it's stack-allocated in `do_markset_autocmd`.  Seems fine for now.